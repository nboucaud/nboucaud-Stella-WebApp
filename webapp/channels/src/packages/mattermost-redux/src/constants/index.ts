// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import Bot from './bot';
import Emoji from './emoji';
import Files from './files';
import General from './general';
import Permissions from './permissions';
import Plugins from './plugins';
import Posts from './posts';
import Preferences from './preferences';
import RequestStatus from './request_status';
import Roles from './roles';
import Stats from './stats';
import Teams from './teams';
import Threads from './threads';
import Users from './users';
import WebsocketEvents from './websocket';

export {
    Bot,
    Emoji,
    Files,
    General,
    Permissions,
    Plugins,
    Posts,
    Preferences,
    RequestStatus,
    Roles,
    Stats,
    Teams,
    Threads,
    Users,
    WebsocketEvents,
};
