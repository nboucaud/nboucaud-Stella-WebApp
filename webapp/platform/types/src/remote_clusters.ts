// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type RemoteClusterInvite = {
    remote_id: string;
    remote_team_id: string;
    site_url: string;
    token: string;
};

export type RemoteClusterAcceptInvite = {
    name: string;
    display_name: string;
    invite: string;
    password: string;
}

export type RemoteClusterPatch = Pick<RemoteCluster, 'display_name'>

export function isRemoteClusterPatch(x: Partial<RemoteClusterPatch>): x is RemoteClusterPatch {
    return x.display_name !== undefined && x.display_name !== '';
}

export type RemoteCluster = {
    remote_id: string;
    remote_team_id: string;
    name: string;
    display_name: string;
    site_url: string;
    create_at: number;
    last_ping_at: number;
    token: string;
    remote_token?: string;
    topics: string;
    creator_id: string;
    plugin_id: string;
    options: number;
}

export type RemoteClusterWithPassword = RemoteCluster & {
    password: string;
}
