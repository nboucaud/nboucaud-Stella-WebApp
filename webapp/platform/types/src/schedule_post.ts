// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {Draft} from './drafts';

export type SchedulingInfo = {
    scheduled_at: number;
    processed_at?: number;
    error_code?: string;
}

export type ScheduledPost = Omit<Draft, 'delete_at'> & SchedulingInfo & {
    id: string;
}

export type ScheduledPostsState = {
    byTeamId: {[teamId: string]: ScheduledPost[]};
}
