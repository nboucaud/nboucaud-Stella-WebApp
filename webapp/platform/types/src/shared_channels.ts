// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type SharedChannel = {
    id: string;
    team_id: string;
    home: boolean;
    readonly: boolean;
    name: string;
    display_name: string;
    purpose: string;
    header: string;
    creator_id: string;
    create_at: number;
    update_at: number;

    /** Omitted when when {@link SharedChannel.home} is `false` */
    remote_id?: string;
};

export type SharedChannelRemote = {
    id: string;
    channel_id: string;
    creator_id: string;
    create_at: number;
    update_at: number;
    is_invite_accepted: boolean;
    is_invite_confirmed: boolean;
    remote_id: string;
    last_post_update_at: number;
    last_post_id: string;
    last_post_create_at: number;
    last_post_create_id: string;
}
