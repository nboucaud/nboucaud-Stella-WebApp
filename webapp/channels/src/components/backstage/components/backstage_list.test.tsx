// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {render, screen} from '@testing-library/react';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {wrapIntl} from '@mattermost/components/src/testUtils';

import BackstageList from './backstage_list';

describe('components/backstage/components/BackstageList', () => {
    test('Should have the browsing hint', () => {
        render(
            wrapIntl(
                <BackstageList
                    header={
                        <FormattedMessage
                            id='installed_incoming_webhooks.header'
                            defaultMessage='Installed Incoming Webhooks'
                        />
                    }
                    hintText={
                        <FormattedMessage
                            id='installed_incoming_webhooks.hint'
                            defaultMessage='Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".'
                        />
                    }
                    loading={false}
                >
                    {() => {
                        return [[], false];
                    }}
                </BackstageList>,
            ),
        );

        const header = screen.getByText('Installed Incoming Webhooks');
        expect(header).toBeInTheDocument();

        const hint = screen.getByText(/Search by title of the/);
        expect(hint).toBeInTheDocument();
    });

    test("Shouldn't have the browsing hint", () => {
        render(
            wrapIntl(
                <BackstageList
                    header={
                        <FormattedMessage
                            id='installed_incoming_webhooks.header'
                            defaultMessage='Installed Incoming Webhooks'
                        />
                    }
                    loading={false}
                >
                    {() => {
                        return [[], false];
                    }}
                </BackstageList>,
            ),
        );
        const header = screen.getByText('Installed Incoming Webhooks');
        expect(header).toBeInTheDocument();

        const hint = screen.queryByText(/Search by title of the/);
        expect(hint).toBeNull();
    });

    test('Should match snapshot with browsing hint', () => {
        const container = render(
            wrapIntl(
                <BackstageList
                    header={
                        <FormattedMessage
                            id='installed_incoming_webhooks.header'
                            defaultMessage='Installed Incoming Webhooks'
                        />
                    }
                    hintText={
                        <FormattedMessage
                            id='installed_incoming_webhooks.hint'
                            defaultMessage='Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".'
                        />
                    }
                    loading={false}
                >
                    {() => {
                        return [[], false];
                    }}
                </BackstageList>,
            ),
        );
        expect(container).toMatchSnapshot();
    });

    test('Should match snapshot without browsing hint', () => {
        const container = render(
            wrapIntl(
                <BackstageList
                    header={
                        <FormattedMessage
                            id='installed_incoming_webhooks.header'
                            defaultMessage='Installed Incoming Webhooks'
                        />
                    }
                    loading={false}
                >
                    {() => {
                        return [[], false];
                    }}
                </BackstageList>,
            ),
        );
        expect(container).toMatchSnapshot();
    });
});
