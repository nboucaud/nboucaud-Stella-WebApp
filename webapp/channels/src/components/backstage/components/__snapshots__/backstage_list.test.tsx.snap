// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/backstage/components/BackstageList Should match snapshot with browsing hint 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="backstage-content"
      >
        <div
          class="backstage-header"
        >
          <h1>
            Installed Incoming Webhooks
          </h1>
        </div>
        <div
          class="backstage-filters"
        >
          <div
            class="backstage-filter__search"
          >
            <i
              class="fa fa-search"
              title="Search Icon"
            />
            <input
              class="form-control"
              id="searchInput"
              placeholder="Search"
              style="flex-grow: 0; flex-shrink: 0;"
              type="search"
              value=""
            />
          </div>
        </div>
        <span
          class="backstage-list__hint"
        >
          Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".
        </span>
        <span
          class="backstage-list__help"
        />
        <div
          class="backstage-list"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="backstage-content"
    >
      <div
        class="backstage-header"
      >
        <h1>
          Installed Incoming Webhooks
        </h1>
      </div>
      <div
        class="backstage-filters"
      >
        <div
          class="backstage-filter__search"
        >
          <i
            class="fa fa-search"
            title="Search Icon"
          />
          <input
            class="form-control"
            id="searchInput"
            placeholder="Search"
            style="flex-grow: 0; flex-shrink: 0;"
            type="search"
            value=""
          />
        </div>
      </div>
      <span
        class="backstage-list__hint"
      >
        Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".
      </span>
      <span
        class="backstage-list__help"
      />
      <div
        class="backstage-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/backstage/components/BackstageList Should match snapshot without browsing hint 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="backstage-content"
      >
        <div
          class="backstage-header"
        >
          <h1>
            Installed Incoming Webhooks
          </h1>
        </div>
        <div
          class="backstage-filters"
        >
          <div
            class="backstage-filter__search"
          >
            <i
              class="fa fa-search"
              title="Search Icon"
            />
            <input
              class="form-control"
              id="searchInput"
              placeholder="Search"
              style="flex-grow: 0; flex-shrink: 0;"
              type="search"
              value=""
            />
          </div>
        </div>
        <span
          class="backstage-list__hint"
        />
        <span
          class="backstage-list__help"
        />
        <div
          class="backstage-list"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="backstage-content"
    >
      <div
        class="backstage-header"
      >
        <h1>
          Installed Incoming Webhooks
        </h1>
      </div>
      <div
        class="backstage-filters"
      >
        <div
          class="backstage-filter__search"
        >
          <i
            class="fa fa-search"
            title="Search Icon"
          />
          <input
            class="form-control"
            id="searchInput"
            placeholder="Search"
            style="flex-grow: 0; flex-shrink: 0;"
            type="search"
            value=""
          />
        </div>
      </div>
      <span
        class="backstage-list__hint"
      />
      <span
        class="backstage-list__help"
      />
      <div
        class="backstage-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
